src/sentry/models/notificationsetting.py

```diff
@@ -118,6 +118,12 @@ class Meta:
                 "type",
             ),
         )
+        constraints = [
+            models.CheckConstraint(
+                check=models.Q(team_id__isnull=False) | models.Q(user_id__isnull=False),
```

Comment:
Can both of these be true? Or should this be `Q(team_id__isnull=False, user_id__isnull=True) | Q(team_id__isnull=True, user_id__isnull=False)`?